pkg/frontend/frontend.go

```diff
@@ -124,6 +125,7 @@ func NewFrontend(ctx context.Context,
 	clusterm metrics.Emitter,
 	aead encryption.AEAD,
 	hiveClusterManager hive.ClusterManager,
+	hiveSyncSetManager hive.SyncSetManager,
```

Comment:
Would it make more sense to add this using the builder pattern or something similar:https://medium.com/@reetas/clean-ways-of-adding-new-optional-fields-to-a-golang-struct-99ae2fe9719d?

What do you think?

Reply:
the rational for creating a new interface is bcoz the syncsets are not cluster scoped and hive instance scoped, so its cleaner to have a new interface which implements syncset functions.